### PR TITLE
feat: Add llama_index_sever

### PR DESCRIPTION
**Pull request by leehuwuj**
## Summary by CodeRabbit

- **New Features**
  - Introduced an automated release and testing pipeline for smoother versioning and reliable publishing.
  - Enhanced the server with interactive chat and workflow management, including real-time streaming responses and automatic UI setup.
  - Added new capabilities for file handling, document generation (PDF/HTML), sandboxed code execution, and tool-assisted query functions.
  - Implemented a new class for managing event streaming and callbacks within the chat functionality.
  - Introduced a new service for suggesting follow-up questions based on chat history.

- **Documentation**
  - Provided comprehensive guides for installation, usage, and API interaction, including detailed instructions for the LlamaIndex Server.

- **Chores**
  - Updated project configuration and ignore settings for a cleaner development workflow.

- **Tests**
  - Expanded test coverage to ensure stability across chat, file, and tool functionalities, including unit tests for new features and edge cases.

### CODE REVIEW

**Review by marcusschiesser — COMMENTED** [review]
.

File: llama-index-server/llama_index/server/chat_ui.py [hunk 1]
```diff
@@ -5,8 +5,7 @@
 
 import requests
 
-# TODO: Update to npm registry once we have a release of @llamaindex/server
-CHAT_UI_VERSION = "1780"
+CHAT_UI_VERSION = "0.0.2"
```
Comment (marcusschiesser) [inline llama-index-server/llama_index/server/chat_ui.py:8]: @leehuwuj need to update this with latest before release

**Review by marcusschiesser — APPROVED** [review]
